Sources/Account/Account+SCIOperations.swift

```diff
@@ -0,0 +1,169 @@
+                serialQueue.async {
+                    completion(.failure(.requiresBlockVersion3(
+                        "Block version must be > 3 for SCI support")))
+                }
+                return false
+            }
+            return true
+        }
+
+        private func verifyAmountIsNonZero(
+            _ amount: Amount,
+            _ actionDescription: String,
+            _ completion: @escaping (
+                Result<SignedContingentInput, SignedContingentInputCreationError>) -> Void
+        ) -> Bool {
+            guard amount.value > 0 else {
+                let errorMessage = "createSignedContingentInput failure: " +
+                    "Cannot \(actionDescription) 0 \(amount.tokenId)"
+                logger.error(errorMessage, logFunction: false)
+                serialQueue.async {
+                    completion(.failure(.invalidInput(errorMessage)))
+                }
+                return false
+            }
+            return true
+        }
+
+        private func logTxOuts(_ txOuts: [KnownTxOut], _ message: String) {
+            logger.info(
+                "\(message): " +
+                    """
+                        0x\(redacting: txOuts.map {
+                            $0.publicKey.hexEncodedString()
+                        })
+                    """,
+                logFunction: false)
+        }
+
+        private func logUnspentTxOuts(
+            _ recipient: PublicAddress,
+            _ amountToSend: Amount,
+            _ amountToReceive: Amount,
+            _ txOuts: [KnownTxOut]
+        ) {
+            logger.info(
+                "Creating signed contingent input to recipient: \(redacting: recipient), " +
+                    "amountToSend: \(redacting: amountToSend), " +
+                    "amountToRecieve: \(redacting: amountToReceive), " +
+                    "unspentTxOutValues: \(redacting: txOuts.map { $0.value })",
+                logFunction: false)
+        }
+
+        func createSignedContingentInput(
+            to recipient: PublicAddress,
+            memoType: MemoType,
+            amountToSend: Amount,
+            amountToReceive: Amount,
+            completion: @escaping (
+                Result<SignedContingentInput, SignedContingentInputCreationError>
+            ) -> Void
+        ) {
+            let functionName = "createSignedContingentInput"
+
+            guard verifyAmountIsNonZero(amountToSend, "send", completion) else {
+                return
+            }
+
+            guard verifyAmountIsNonZero(amountToReceive, "receive", completion) else {
+                return
+            }
+
+            // get all unspent txOuts
+            let (unspentTxOuts, ledgerBlockCount) =
+            account.readSync {
+                ($0.unspentTxOuts(tokenId: amountToSend.tokenId), $0.knowableBlockCount)
+            }
+            logUnspentTxOuts(recipient, amountToSend, amountToReceive, unspentTxOuts)
```

Comment:
This feels like debug code, so it shouldn't be merged to master.

Reply:
looking at the function, some of the logging could be useful, Id inline it instead of having it as a function